### PR TITLE
feat: Implement incoming requests

### PR DESCRIPTION
**Pull request by Elyniss**
Merge after #815

### CODE REVIEW

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
 
-internal class OnDeleteRequestUseCase(private val storesRepository: StoresStorageRepository, private val accountsRepository: AccountsStorageRepository) {
-    suspend operator fun invoke(params: SyncParams.DeleteParams, request: WCRequest) {
-        TODO()
+internal class OnDeleteRequestUseCase(
+    private val storesRepository: StoresStorageRepository,
+    private val logger: Logger,
+    private val jsonRpcInteractor: JsonRpcInteractorInterface,
+) {
+    suspend operator fun invoke(params: SyncParams.DeleteParams, request: WCRequest, events: MutableSharedFlow<EngineEvent>) {
```
Comment (TalhaAli00) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt:22]: I'm unsure if we should use this pattern of passing the MutableSharedFlow to emit a value vs having an internal SharedFlow and then we combine in the engine. Curious what everyone else's opinions are

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnSetRequestUseCase.kt [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
-
-        // get account by request.topic
-        // accountsRepository.getAccountByTopic() -> implement
-//        validateAccountId(accountId) { error -> return@supervisorScope onFailure(error) }
+internal class OnSetRequestUseCase(
+    private val storesRepository: StoresStorageRepository,
+    private val logger: Logger,
+    private val jsonRpcInteractor: JsonRpcInteractorInterface,
+) {
+    suspend operator fun invoke(params: SyncParams.SetParams, request: WCRequest, events: MutableSharedFlow<EngineEvent>) {
```
Comment (TalhaAli00) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnSetRequestUseCase.kt:22]: same comment here about SharedFlow

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt (hunk 1 above)
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt:22]: I'm not gonna lie, I was very unsure when I passed this as argument. Curious to hear your opinions

Comment (jakubuid) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt:22]: Oh def internal SharedFlow that is combined later in the engine

Comment (jakubuid) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt:22]: This usecase should emit its own flow as JsonRpcInteractor does. Then in the engine, all flows from different sources that are emitted as EngineEvent should be combined

Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/OnDeleteRequestUseCase.kt]: Check out 
[`ba26eb1` (#816)](https://github.com/WalletConnect/WalletConnectKotlinV2/pull/816/commits/ba26eb12ae95fd9b65965ebf0eb17b04fa475ad2)

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**